rids/iocs/iocs.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 Jigsaw Operations LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Combined IOC feed parser interface for the different formats of IOCs."""
+
+
+import collections
+from dataclasses import dataclass
+from types import Set
+
+from rids.iocs import allowed_sni_port
+from rids.iocs import bad_ip_list
+from rids.monitors.ip_monitor import IpRule
+from rids.monitors.tls_monitor import TlsRule
+
+
+_IOC_PARSERS = {
+  "BAD_IP_LIST": bad_ip_list.BadIpAddresses,
+  "ALLOWED_SNI_PORT": allowed_sni_port.AllowedEndpoints,
+}
+
+
+def fetch_iocs(rids_config):
+  """Parses the IOC indicated by `ioc_config` into a RuleSet instance.
+  
+  Args:
+    rids_config: { 'iocs': list[dict{'name', 'url', 'format'}] }
+
+  Returns:
+    RuleSet representing all IOCs in the config.
+  """
+  ioc_sources = rids_config.get('ioc_sources', None)
+  ruleset = RuleSet()
+  if not ioc_sources:
+    return ruleset
+
+  for ioc_config in ioc_sources:
+    format = ioc_config['format']
+    if format not in _IOC_PARSERS:
+      raise ValueError('Unrecognized IOC feed format "{format}"')
+
+    ioc_source = _IOC_PARSERS[format](ioc_config)
+    ioc_source.provide_rules(ruleset)
+
+  return ruleset
+
+
+@dataclass
+class RuleSet:
```

Comment:
As per our discussion: Move the IP logic to some `IpMatcher` in a separate IP file, and the TLS logic to a `TlsMatcher` in a tls file. This class can then hold those two objects, but code depending on one of those logics won't need a full RuleSet.